llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7954,9 +7954,9 @@ bool VPRecipeBuilder::getScaledReductions(
   auto CollectExtInfo = [this, &Exts, &ExtOpTypes,
                          &ExtKinds](SmallVectorImpl<Value *> &Ops) -> bool {
     for (const auto &[I, OpI] : enumerate(Ops)) {
-      auto *CI = dyn_cast<ConstantInt>(OpI);
-      if (I > 0 && CI &&
-          canConstantBeExtended(CI, ExtOpTypes[0], ExtKinds[0])) {
+      const APInt *C;
```

Comment:
I guess if all you care about is passing a APInt to `canConstantBeExtended` you could equally achieve the same with the original code:

```
      auto *CI = dyn_cast<ConstantInt>(OpI);
      if (I > 0 && CI &&
          canConstantBeExtended(CI->getValue(), ExtOpTypes[0], ExtKinds[0])) {
```

but I'm happy either way.